### PR TITLE
Replace token file with Google auth in MapBuilder.

### PR DESCRIPTION
**Pull request by stevewolter**
This allows Cartographer to use normal Application Default Credentials
(ADC) instead of a hand-generated format.

Google-internally, we'll only use ADC going forward. At your option, I
can keep the old code for the token files around if you think it would
be generally useful. Many OSS programs follow the same approach.

### CODE REVIEW

File: cartographer/cloud/internal/local_trajectory_uploader.cc [hunk 1]
```diff
@@ -108,19 +107,15 @@ class LocalTrajectoryUploader : public LocalTrajectoryUploaderInterface {
...
-    auto call_creds = async_grpc::TokenFileCredentials(
-        token_file_path, std::chrono::seconds(kTokenRefreshIntervalInSeconds));
-    channel_creds =
-        grpc::CompositeChannelCredentials(channel_creds, call_creds);
-  }
+      enable_google_auth
+          ? grpc::GoogleDefaultCredentials()
+          : enable_ssl_encryption
+                ? ::grpc::SslCredentials(::grpc::SslCredentialsOptions())
+                : ::grpc::InsecureChannelCredentials();
```
Comment (cschuet) [inline cartographer/cloud/internal/local_trajectory_uploader.cc]: Wow :) Can you put some brackets to clarify the evaluation order?

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by stevewolter** [comment]
Sure, I sent you a PR for removing TokenFileCredentials in async_grpc.

File: cartographer/cloud/internal/local_trajectory_uploader.cc (hunk 1 above)
Comment (stevewolter) [reply cartographer/cloud/internal/local_trajectory_uploader.cc]: Done. It's only one pair of parenthesis, but maybe it makes things easier to read :).

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**